src/H5Tpkg.h

```diff
@@ -45,8 +45,7 @@
 #define H5T_NAMELEN 32
 
 /* Macro to ease detecting "complex" datatypes (i.e. those with base types or fields) */
-#define H5T_IS_COMPLEX(t)                                                                                    \
-    ((t) == H5T_COMPOUND || (t) == H5T_ENUM || (t) == H5T_VLEN || (t) == H5T_ARRAY || (t) == H5T_REFERENCE)
+#define H5T_IS_COMPLEX(t) ((t) == H5T_COMPOUND || (t) == H5T_ENUM || (t) == H5T_VLEN || (t) == H5T_ARRAY)
```

Comment:
I believe `H5T_REFERENCE` was only added to this macro to make the logic in `H5T_set_loc` work correctly for reference types. Reference types are treated as atomic types, but would be excluded when checking a datatype with the `H5T_IS_ATOMIC` macro due to the `!H5T_IS_COMPLEX` check.